### ISSUE TITLE
I like the idea of this class, but implement LRU cache here once again is redundant 

### ISSUE THREAD

**Issue by irenkamalova** [Ellena557/LRU-cache#7]
https://github.com/Ellena557/LRU-cache/blob/e4baeccd54a7334dcc8cdeb33e95fe8e396a47a8/lru-cache/src/main/java/ru/ellen/example/CachedAlgorithmCounter.java#L8

**Comment by Ellena557** [comment]
Yes, it is. The task seemed to be a little bit ambiguous: were we supposed to implement an interface or just to show how the logic works. I decided to play it safe and wrote 2 versions:)
